lib/rules/dot-location.js

```diff
@@ -54,29 +54,36 @@ module.exports = {
          */
         function checkDotLocation(obj, prop, node) {
             const dot = sourceCode.getTokenBefore(prop);
-            const textBeforeDot = sourceCode.getText().slice(obj.range[1], dot.range[0]);
+
+            /* istanbul ignore if */
+            if (astUtils.isNotDotToken(dot)) {
```

Comment:
What's the purpose of this conditional? Is it just a sanity check?

Reply:
Yes, it's a bit strange now. 

This should never happen, I couldn't find an example in proposals, typescript etc. It's basically inherited from the previous version, the purpose was most likely to check for `[]` (computed key), which isn't necessary now. 

But, it was also a check for an unknown syntax, so I wasn't sure should it stay or not. 

Anyway, I completely agree that the purpose of this line is not clear to a reader, and it's better to do something about it. Should I remove the check, or add a comment?

Reply:
I've added the comment.

If it would be better, I can remove the whole conditional, too.

Reply:
Unless we can find a test case where this guard is necessary, I’d be more inclined to remove it for future maintainers’ sake.